data_collection/gazette/spiders/pa/pa_parauapebas.py

```diff
@@ -0,0 +1,47 @@
+import datetime as dt
+
+import scrapy
+from dateparser import parse
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class PaParauapebasSpider(BaseGazetteSpider):
+    TERRITORY_ID = "1505536"
+    name = "pa_parauapebas"
+    start_urls = ["http://apps.ioepa.com.br/Parauapebas/Busca"]
+    start_date = dt.date(2022, 1, 4)
+
+    def parse(self, response):
+        ano_atual = dt.datetime.now().year
+        for year in [2022, ano_atual]:
+            for month in range(1, 13):
+                for day in range(1, 32):
+                    if len(str(day)) < 2:
+                        day = "".join(("0", str(day)))
+                    if len(str(month)) < 2:
+                        month = "".join(("0", str(month)))
+                    data = f"{year}-{month}-{day}"
+                    url = f"{response.url}?data={data}"
+                    yield scrapy.Request(
+                        url,
+                        callback=self.parse_gazette,
+                        cb_kwargs={"data": data},
+                    )
```

Comment:
Aqui temos dois comentários importantes:
1. Para gerar datas em um dado intervalo a função `rrule` da biblioteca [`dateutil`](https://dateutil.readthedocs.io/en/stable/) é uma mão na roda gigante e evita gerarmos datas inválidas (como 31 de fevereiro).
2. É obrigatório implementar o filtro por datas por meio dos atributos `self.start_date` e `self.end_date`. É importante dar uma lida [nessa página da documentação](https://docs.queridodiario.ok.org.br/pt/latest/escrevendo-um-novo-spider.html#filtro-por-data).
```suggestion
        for date in rrule(DAILY, dtstart=self.start_date, until=self.end_date):  # fazer uso do rrule para não gerar datas inválidas
                                                                                 # fazer uso de start_date e end_date é obrigatório na escrita do raspador
            yield scrapy.Request(
                f"{response.url}?data={date.date().isoformat()}",
                callback=self.parse_gazette,
                cb_kwargs={"data": date.date()},
```